mino/minogrpc/overlay_test.go

```diff
@@ -191,3 +197,15 @@ func (t testFailHandler2) Stream(out mino.Sender, in mino.Receiver) error {
 	require.EqualError(t.t, err, "")
 	return nil
 }
+
+type goodEncoder struct {
```

Comment:
you can use `encoding.ProtoEncoder` that implements encoding.ProtoMarshaler instead of the fake ones. Then keep your function implementation to ignore.